nexus/types/src/deployment/planning_input.rs

```diff
@@ -240,6 +246,15 @@ impl PlanningInput {
         self.policy.oximeter_read_policy.mode.single_node_enabled()
     }
 
+    /// ID of the currently running Nexus zone
+    pub fn current_nexus_zone_id(&self) -> Option<OmicronZoneUuid> {
+        self.current_nexus_zone_id
+    }
+
+    pub fn set_current_nexus_zone_id(&mut self, id: OmicronZoneUuid) {
+        self.current_nexus_zone_id = Some(id);
+    }
```

Comment:
nit: I've thought of these types like `PlanningInput` being immutable and `PlanningInputBuilder` being the mutable version.  What do you think of having callers do this:

```rust
let mut new_builder = planning_input.into_builder();
new_builder.set( ... )
let planning_input = new_builder.build();
```

It's a little more verbose but I feel like preserves the nice property that: when you're modifying it, you're working with a builder.  The planning input itself remains immutable.  (Again, take it or leave it.)